### ISSUE TITLE
Call to undefined method BitWasp\Bitcoin\Crypto\EcAdapter\Impl\PhpEcc\Key\PublicKey::getAddress()

### ISSUE THREAD

**Issue by tkalfigo** [dan-da/hd-wallet-addrs#8]
On Ubuntu with PHP version:
```
PHP 7.2.5-0ubuntu0.18.04.1 (cli) (built: May  9 2018 17:21:02) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.5-0ubuntu0.18.04.1, Copyright (c) 1999-2018, by Zend Technologies
```

Executing README example:
```
$ ./hd-wallet-addrs.php -g --xpub=xpub6BfKpqjTwvH21wJGWEfxLppb8sU7C6FJge2kWb9315oP4ZVqCXG29cdUtkyu7YQhHyfA5nt63nzcNZHYmqXYHDxYo8mm1Xq1dAC7YtodwUR --cols=type,abspath,relpath,addr --derivation=bip44 --logfile=/tmp/log.txt
```

results in error:

```
Uncaught Exception. code: 0, message: Call to undefined method BitWasp\Bitcoin\Crypto\EcAdapter\Impl\PhpEcc\Key\PublicKey::getAddress()
/home/user/hd-wallet-addrs/lib/walletaddrs.class.php : 187

Stack Trace:
#0 /home/user/hd-wallet-addrs/lib/walletaddrs.class.php(110): walletaddrs->discover_addrs(Object(BitWasp\Bitcoin\Key\Deterministic\HierarchicalKey))
#1 /home/user/hd-wallet-addrs/lib/walletaddrs.class.php(54): walletaddrs->discover_wallet_addrs_single('xpub6BfKpqjTwvH...')
#2 /home/user/hd-wallet-addrs/hd-wallet-addrs.php(35): walletaddrs->discover_wallet_addrs(Array)
#3 /home/user/hd-wallet-addrs/hd-wallet-addrs.php(19): main(Array)
#4 {main}
WARNING: ALERTS_MAIL_TO not defined in config.  alert not sent with subject: [my-pc] [hd-wallet-addrs.php] Uncaught exception! [pid: 12624]
```

**Comment by dan-da** [comment]
Thanks for the report.   fixed in 65e1683ab1dd16f69617c6be149e99fe63a0fb53

To use you would just need to:
```
$ cd hd-wallet-addrs
$ git pull
$ composer update
```

bitwasp API has changed in recent releases.   I locked composer.json to an older version of bitwasp that is known to work.    Later I will see about updating the code to work with recent bitwasp.

**Comment by dan-da** [comment]
compat with bitwasp/bitcoin latest release v0.0.35 in 65b36cf2aa5763a4b721e8659bf29cb9d7697365.  closing.
